### PR TITLE
feat: [CDS-72845]: handled ECS override types

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger standards

**Review by tan-nhu — DISMISSED** [review]

**Comment by kalyansai-harness** [comment]
@pankajsingh-harness @chetan-harness could you please review this PR.

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts [hunk 1]
```diff
@@ -123,11 +166,19 @@ export const TASOverrideManifests = [
...
+  OverrideManifests.EcsScalingPolicyDefinition
+]
+
 export function getAllowedOverrideManifests({ NG_SVC_ENV_REDESIGN = false }): OverrideManifestTypes[] {
   let overrideManifests = [...AllowedManifestOverrideTypes]
 
   if (NG_SVC_ENV_REDESIGN) {
     overrideManifests = overrideManifests.concat(TASOverrideManifests)
   }
+  overrideManifests = overrideManifests.concat(ECSOverrideManifests)
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:182]: directly add this to `overrideManifests`. No need to contact later.

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts [hunk 2]
```diff
@@ -123,11 +163,19 @@ export const TASOverrideManifests = [
...
 
+export const ECSOverrideManifests = [
+  OverrideManifests.EcsTaskDefinition,
+  OverrideManifests.EcsServiceDefinition,
+  OverrideManifests.EcsScalableTargetDefinition,
+  OverrideManifests.EcsScalingPolicyDefinition
+]
+
 export function getAllowedOverrideManifests({ NG_SVC_ENV_REDESIGN = false }): OverrideManifestTypes[] {
   let overrideManifests = [...AllowedManifestOverrideTypes]
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:174]:
```suggestion
  let overrideManifests = [...AllowedManifestOverrideTypes, ...ECSOverrideManifests]
```

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts [hunk 3]
```diff
@@ -123,11 +163,19 @@ export const TASOverrideManifests = [
...
 
   if (NG_SVC_ENV_REDESIGN) {
     overrideManifests = overrideManifests.concat(TASOverrideManifests)
   }
+  overrideManifests.concat(ECSOverrideManifests)
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:179]: Remove this after the change in line 174

Comment (kalyansai-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:179]: @pankajsingh-harness just a question. Is there a specific sequence of overrideManifests required in the array? I mean first all `AllowedManifestOverrideTypes` next is `TASOverrideManifests` and in the end it is `ECSOverrideManifests`

If there is no such sequence is required, then we can make the changes as per your comment.

Comment (sahithikolichala) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:179]: there is no mention of ordering, these can apply in any order

Comment (pankajsingh-harness) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/ServiceOverrides/ServiceManifestOverride/ServiceManifestOverrideUtils.ts:179]: @kalyansai-harness I don't think so. @sahithikolichala @chetan-harness can confirm this sequence ?

**Review by pankajsingh-harness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good Sahithi

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**